clang/runtime/dpct-rt/include/math.hpp.inc

```diff
@@ -42,6 +42,62 @@
 // DPCT_LABEL_END
 
 namespace dpct {
+// DPCT_LABEL_BEGIN|fast_length|dpct
+// DPCT_DEPENDENCY_EMPTY
+// DPCT_CODE
+/// Compute fast_length for variable-length array
+/// \param [in] a The array
+/// \param [in] len Length of the array
+/// \returns The computed fast_length
+inline float fast_length(const float *a, int len) {
+  switch (len) {
+  case 1:
+    return a[0];
+  case 2:
+    return sycl::fast_length(sycl::float2(a[0], a[1]));
+  case 3:
+    return sycl::fast_length(sycl::float3(a[0], a[1], a[2]));
+  case 4:
+    return sycl::fast_length(sycl::float4(a[0], a[1], a[2], a[3]));
+  case 0:
+    return 0;
+  default:
+    float f = 0;
+    for (int i = 0; i < len; ++i)
+      f += a[i] * a[i];
+    return sycl::sqrt(f);
+  }
+}
+// DPCT_LABEL_END
+
+// DPCT_LABEL_BEGIN|length|dpct
+// DPCT_DEPENDENCY_EMPTY
+// DPCT_CODE
+/// Calculate the square root of the input array.
+/// \param [in] a The array pointer
+/// \param [in] len Length of the array
+/// \returns The square root
+template <typename T> inline T length(const T *a, const int len) {
+  if (len < 1)
```

Comment:
how about put if in default, then when 1, 2, 3,4, it don't need exe this branch.